haas/deferred.py

```diff
@@ -51,22 +52,33 @@ def apply_networking():
         network = action.new_network
         if nic.port:
             if network:
-                network_map[nic.port.label] = network.network_id
+                network_id = network.network_id
             else:
-                network_map[nic.port.label] = None
+                network_id = None
+
+            switch = nic.port.owner
+            if switch.label not in switch_sessions:
+                switch_sessions[switch.label] = switch.session()
+            switch_sessions[switch.label].apply_networking(action)
         else:
             logging.getLogger(__name__).warn(
                 'Not modifying NIC %s; NIC is not on a port.' %
                 nic.label)
 
-    # Apply them
-    driver_name = config.cfg.get('general', 'driver')
-    driver = importlib.import_module('haas.drivers.' + driver_name)
-    driver.apply_networking(network_map)
+    # Close all of our sessions:
+    for session in switch_sessions.values():
+        session.disconnect()
 
-    # Then perform the database change and delete them
+    # Then perform the database changes and delete them
     for action in actions:
-        action.nic.network = action.new_network
+        if action.new_network is None:
+            db.query(model.NetworkAttachment)\
+                .filter_by(nic=action.nic, channel=action.channel)\
+                .delete()
```

Comment:
I think `one()` actually fetches the object, whereas `delete` is a method on the query; I don't think that will work at all (the object doesn't have a delete method), and even if it does it will needlessly execute an extra query.